aiidalab_widgets_base/viewers.py

```diff
@@ -1165,13 +1165,15 @@ def __init__(self, bands, **kwargs):
 
 
 @register_viewer_widget("process.workflow.workchain.WorkChainNode.")
-class WorkChainNodeViewerWidget(ipw.HTML):
-    def __init__(self, workchain, **kwargs):
-        self.workchain = workchain
-
-        # Displaying reports only from the selected workchain,
-        # NOT from its descendants
-        report = get_workchain_report(self.workchain, "REPORT", max_depth=1)
+@register_viewer_widget("process.calculation.calcfunction.CalcFunctionNode.")
+@register_viewer_widget("process.workflow.workfunction.WorkFunctionNode.")
+class ProcessNodeViewerWidget(ipw.HTML):
+    def __init__(self, process, **kwargs):
+        self.process = process
+
+        # Displaying reports only from the selected process,
+        # NOT from its descendants.
+        report = get_workchain_report(self.process, "REPORT", max_depth=1)
```

Comment:
Agreed, I was thinking about this. However, in the context of the `ProcessNodesTreeWidget`, the basic information (name of the process and its pk for example) is already displayed in the tree so it felt redundant to add it here as well. But perhaps there are other contexts where this would be useful?

Reply:
Maybe start time, finish time, UUID. Also, keep in mind that the viewer is not only used in the `ProcessNodesTreeWidget`

Reply:
I added the additional metadata, here's how it looks now:

![image](https://user-images.githubusercontent.com/9539441/189689872-6356e32b-e0cb-4efb-a045-6a791e2b9168.png)

Reply:
@yakutovicha feel free to modify the header format to your liking, I was going back and forth, trying to minimize the number of extra lines but keeping it readable.

Reply:
I like the way you made it 👍